File: programming-concepts/logical-operators/python/nand.py
# Raspberry Pi Foundation
# Developed as part of Ada Computer Science
# 
# Usage licensed under CC BY-NC-SA 4.0


def main():
    cold = True
    wet = False

    if not(cold and wet):
        print("It's OK to go outside")
    else:
        print("I'm going to stay inside")

        
# This code will run if this file is executed directly
# (i.e. not called by another program)
if __name__ == '__main__':
    main()

File: programming-concepts/logical-operators/visual-basic/nand.vb
' Raspberry Pi Foundation
' Developed as part of Ada Computer Science
' Usage licensed under CC BY-NC-SA 4.0
' Note: This file is designed to be copied out and compiled on your machine.
' To run this file you need to:
' 1. Copy the contents
' 2. Paste them into the Visual Basic IDE of your choice (Visual Studio, for example)
' 3. Compile the program
' 4. Run the program

Module Program

    ' The Main subroutine is the default entry point for a VB program
    Sub Main()
        Dim cold As Boolean = True
        Dim wet As Boolean = False

        If Not (cold AndAlso wet) Then
            Console.WriteLine("It's OK to go outside")
        Else
            Console.WriteLine("I'm going to stay inside")
        End If
    End Sub

End Module
